shell/platform/android/io/flutter/embedding/engine/systemchannels/KeyEventChannel.java

```diff
@@ -0,0 +1,50 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.systemchannels;
+
+import android.support.annotation.NonNull;
+import android.view.KeyEvent;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.BasicMessageChannel;
+import io.flutter.plugin.common.JSONMessageCodec;
+
+public class KeyEventChannel {
```

Comment:
Javadoc here and below?

Reply:
I'd like to avoid class level javadocs for now on the system channels, unless you know where I can find people that can help write them.  What I did for each channel is just copy over the logic that was previously baked into FlutterView.  I don't actually the scope of what these channels are intended to do.  Do you happen to know?

Reply:
So at a minimum Javadoc should be on all public members of an API. Should these actually be package private instead? Since like you said these really only exist to support `FlutterView`. I don't know the scope of these channels well enough to really document them unfortunately.

https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required